### ISSUE TITLE
private-cluster module doesn't handle node_pools variable if it's set as a list of objects with optional object attributes

### ISSUE THREAD

**Issue by ivankorn** [terraform-google-modules/terraform-google-kubernetes-engine#1568]
### TL;DR

Private cluster module doesn't handle node_pools variable if it's set as a list of objects with optional object attributes.
`node_locations` is not required for `node_pools` variable, but the `private-cluster` module fails on trying to split the null variable 
```
│   on .terraform/modules/gke/modules/private-cluster/cluster.tf line 317, in resource "google_container_node_pool" "pools":
│  317:   node_locations = lookup(each.value, "node_locations", "") != "" ? split(",", each.value["node_locations"]) : null
│     ├────────────────
│     │ while calling split(separator, str)
│     │ each.value["node_locations"] is null
```

### Expected behavior

User should be able to define `node_pools` as list objects with `optional` attributes and the module code should handle it correctly. 

### Observed behavior

the module code doesn't respect optional attribute feature of terraform and does not handle the `null` coming from optional attribute and  fails on trying to split split the null variable

```
│ Error: Invalid function argument
│ 
│   on .terraform/modules/gke/modules/private-cluster/cluster.tf line 317, in resource "google_container_node_pool" "pools":
│  317:   node_locations = lookup(each.value, "node_locations", "") != "" ? split(",", each.value["node_locations"]) : null
│     ├────────────────
│     │ while calling split(separator, str)
│     │ each.value["node_locations"] is null
│ 
│ Invalid value for "str" parameter: argument must not be null.
╵
╷
│ Error: Invalid function argument
│ 
│   on .terraform/modules/gke/modules/private-cluster/cluster.tf line 317, in resource "google_container_node_pool" "pools":
│  317:   node_locations = lookup(each.value, "node_locations", "") != "" ? split(",", each.value["node_locations"]) : null
│     ├────────────────
│     │ while calling split(separator, str)
│     │ each.value["node_locations"] is null
│ 
│ Invalid value for "str" parameter: argument must not be null.
╵
╷
│ Error: Invalid function argument
│ 
│   on .terraform/modules/gke/modules/private-cluster/cluster.tf line 317, in resource "google_container_node_pool" "pools":
│  317:   node_locations = lookup(each.value, "node_locations", "") != "" ? split(",", each.value["node_locations"]) : null
│     ├────────────────
│     │ while calling split(separator, str)
│     │ each.value["node_locations"] is null
│ 
│ Invalid value for "str" parameter: argument must not be null.

```

### Terraform Configuration

```hcl
main.tf

module "gke" {
  source                          = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
  version                         = "23.3.0"
  project_id                      = var.project_id
  name                            = var.environment
  regional                        = true
  region                          = var.region
  zones                           = var.zones
  initial_node_count              = var.gke_cluster_settings.node_count
  node_pools                      = var.gke_node_pools
  remove_default_node_pool        = true
  network_project_id              = var.gke_networking.node_pools.network_project_id
  network                         = var.gke_networking.node_pools.network_name
  subnetwork                      = "${var.project_id}-${lookup(var.gke_networking.node_pools.subnetwork_sfx, var.environment)}-${var.region}"
  ip_range_pods                   = var.gke_networking.node_pools.ip_range.pods
  ip_range_services               = var.gke_networking.node_pools.ip_range.services
  create_service_account          = false
  service_account                 = data.google_service_account.gke_nodes.email
  horizontal_pod_autoscaling      = true
  maintenance_start_time          = var.gke_cluster_settings.maintenance_start_time
  enable_private_nodes            = true
  enable_vertical_pod_autoscaling = true
  cluster_autoscaling             = var.gke_autoscaling
  master_ipv4_cidr_block          = lookup(var.gke_networking.master.ipv4_cidr_block, var.environment)
  kubernetes_version              = var.gke_cluster_settings.versions.kubernetes
  release_channel                 = var.gke_cluster_settings.versions.release_channel
  node_pools_tags = {
    all = local.node_network_tags
  }
  configure_ip_masq    = true
  non_masquerade_cidrs = local.internal_ip_cidr_range
}
```

variables.tf
```
variable "gke_node_pools" {
  description = "List of maps containing node pools"
  type = list(object({
    name               = string
    machine_type       = string
    min_count          = number
    max_count          = number
    node_locations     = optional(string)
    disk_type          = optional(string)
    initial_node_count = optional(number)
    disk_size_gb       = optional(number)
    local_ssd_count    = optional(number)
    autoscaling        = optional(bool)
    auto_repair        = optional(bool)
    auto_upgrade       = optional(bool)
    spot               = optional(bool)
  }))
}
```

staging.tfvars
```
gke_node_pools = [
  {
    name         = "default-node-pool"
    machine_type = "n1-standard-2"
    autoscaling  = true
    auto_upgrade = true
    min_count    = 0
    max_count    = 10
  },
  {
    name         = "standard"
    machine_type = "n2-standard-16"
    autoscaling  = true
    auto_upgrade = true
    min_count    = 1
    max_count    = 10
  },
  {
    name         = "highmem"
    machine_type = "n2-highmem-16"
    autoscaling  = true
    auto_upgrade = true
    min_count    = 0
    max_count    = 10
  }
]

pubsub_topics = [
  "evaluate-changed-betty-articles"
]
```
```


### Terraform Version

```sh
$ terraform -version
Terraform v1.3.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.53.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.18.0
+ provider registry.terraform.io/hashicorp/random v3.4.3
```
```


### Additional information

_No response_
